Restivus/Http.swift

```diff
@@ -17,7 +17,7 @@ public enum HTTPError: Error {
     unexpectedResponse(URLResponse),
 
     ///The HTTPURLResponse has a status code other than 2xx
-    unsuccessfulResponse(HTTPURLResponse),
+    unsuccessfulResponse(HTTPURLResponse, Data?),
```

Comment:
I think it might be worth adding the Data? To other HTTPErrors as well, such as .unexpectedResponse and .other